### DISCUSSION TITLE
Partial addition in scalarMulBySeed

### DISCUSSION THREAD

**Discussion by hussein-aitlahcen** [Consensys/gnark#1058] [Q&A] [answered]
Hi,

[In the scalarMulBySeed](https://github.com/Consensys/gnark/blob/1cc2d6ae4aced20f5201e84458e66e27dcdf52b6/std/algebra/emulated/sw_bn254/g2.go#L114) function we use partial addition; are we guaranteed not to hit the infinity point in this operation?

**Comment by ivokub** [comment]
cc. @yelhousni

**Reply by yelhousni** [reply, marked as answer]
No it is an addition chain that goes `2q, 3q, 4q, 6q, 15q, 18q ...`. So unless you start with the infinity point I don't think you can hit incomplete paths.

**Reply by hussein-aitlahcen** [reply]
Thank you
